src/LinearSolverTypes.f90

```diff
@@ -63,6 +63,14 @@ MODULE LinearSolverTypes
 #ifdef FUTILITY_HAVE_Trilinos
 USE ForTeuchos_ParameterList
 #endif
+
+#ifdef FUTILITY_HAVE_PETSC
+#include <petscversion.h>
+#if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>6))
```

Comment:
So would it be possible to process this into its own preproc symbol during configuration?  I can see already that this combination (and my recommended addition above) are going to show up in a bunch of places.

It's definitely not a must, but it would look a little cleaner.

Reply:
there are at least a few minor version variants here and there